datasetviewer/preview/PreviewPresenter.py

```diff
@@ -0,0 +1,47 @@
+from datasetviewer.presenter.SubPresenter import SubPresenter
+from datasetviewer.mainview.interfaces.MainViewPresenterInterface import MainViewPresenterInterface
```

Comment:
This illustrates the conceptual problem I highlighted in my slides, but have also discussed with you...
You have now added `set_data` to the `MainViewPresenterInterface`. Does the `PreviewPresenter` ever need to call `set_data`...probably would be dangerous if it did. Therefore sub-presenters do not use the same interface. Each `sub-presenter` may need it's own bespoke interface to `MainViewPresenter`. You can create a new issue to fix this as a follow up.